velox/exec/tests/AggregationTest.cpp

```diff
@@ -611,6 +676,89 @@ TEST_F(AggregationTest, partialAggregationMemoryLimit) {
       .assertResults("SELECT c0, count(1) FROM tmp GROUP BY 1");
 }
 
+TEST_F(AggregationTest, spill) {
+  using core::QueryConfig;
+  constexpr int32_t kNumDistinct = 200000;
+  constexpr int64_t kMaxBytes = 24LL << 20; // 24 MB
+  filesystems::registerLocalFileSystem();
+  rng_.seed(1);
+  rowType_ = ROW({"c0", "c1", "a"}, {INTEGER(), VARCHAR(), VARCHAR()});
+  // The input batch has kNumDistinct distinct keys. The repeat count of a key
+  // is given by min(1, (k % 100) - 90). The batch is repeated 3 times, each
+  // time in a different order.
+  RowVectorPtr rows = std::static_pointer_cast<RowVector>(
+      BaseVector::create(rowType_, kNumDistinct, pool_.get()));
+  folly::F14FastSet<uint64_t> order1;
+  folly::F14FastSet<uint64_t> order2;
+  folly::F14FastSet<uint64_t> order3;
+  auto c0 = rows->childAt(0)->as<FlatVector<int32_t>>();
+  c0->resize(kNumDistinct);
+  auto c1 = rows->childAt(1)->as<FlatVector<StringView>>();
+  c1->resize(kNumDistinct);
+  int32_t totalCount = 0;
+  for (int32_t i = 0; i < kNumDistinct; ++i) {
+    c0->set(i, i);
+    std::string str = fmt::format("{}{}", i, i);
+    c1->set(i, StringView(str));
+    auto numRepeats = std::max(1, (i % 100) - 90);
+    // We make random permutations of the data by adding the indices into a set
+    // with a random 6 high bits followed by a serial number. These are inlined
+    // in the F14FastSet in an order that depends on the hash number.
+    for (auto j = 0; j < numRepeats; ++j) {
+      ++totalCount;
+      insertRandomOrder(i, totalCount, order1);
+      insertRandomOrder(i, totalCount, order2);
+      insertRandomOrder(i, totalCount, order3);
+    }
+  }
+  std::vector<RowVectorPtr> batches;
+  makeBatches(rows, order1, batches);
+  makeBatches(rows, order2, batches);
+  makeBatches(rows, order3, batches);
+  auto tempDirectory = exec::test::TempDirectoryPath::create();
+  CursorParameters params;
+  std::unordered_map<std::string, std::string> configStrings;
+  configStrings[QueryConfig::kMaxPartialAggregationMemory] = "1600000";
+  configStrings[QueryConfig::kSpillPath] = tempDirectory->path;
+  std::shared_ptr<Config> config =
+      std::make_shared<core::MemConfig>(configStrings);
+  params.queryCtx = core::QueryCtx::createForTest(std::move(config));
+  params.queryCtx->pool()->setMemoryUsageTracker(
+      velox::memory::MemoryUsageTracker::create(kMaxBytes, 0, kMaxBytes));
+
+  params.planNode = PlanBuilder()
+                        .values(batches)
+                        .partialAggregation({"c0", "c1"}, {"array_agg(a)"})
```

Comment:
It doesn't feel sufficient to test just one array_agg aggregate function. It is also not clear whether the test covers paths with nulls. Perhaps, we could extend AggregationTestBase::testAggregations to include a run with spilling.